uplink/metainfo/client.go

```diff
@@ -316,14 +334,21 @@ func convertBucketToProtoRequest(bucket storj.Bucket) pb.BucketCreateRequest {
 			CipherSuite: pb.CipherSuite(bucket.DefaultEncryptionParameters.CipherSuite),
 			BlockSize:   int64(bucket.DefaultEncryptionParameters.BlockSize),
 		},
-	}
+	}, nil
 }
 
-func convertProtoToBucket(pbBucket *pb.Bucket) storj.Bucket {
+func convertProtoToBucket(pbBucket *pb.Bucket) (bucket storj.Bucket, err error) {
 	defaultRS := pbBucket.GetDefaultRedundancyScheme()
 	defaultEP := pbBucket.GetDefaultEncryptionParameters()
+	var partnerID uuid.UUID
+	err = partnerID.UnmarshalJSON(pbBucket.GetPartnerId())
```

Comment:
many places in the code use this function `bytesToUUID()`. Not sure if that is better than this unmarshalling

Reply:
I totally agree with you on that. I modified it to use uuid package for the files and functions I modified. I see there were 37reference all over satellite. My recommendation, is to do incremental changes as we know it works for now.